chaincode/src/contracts/DexV3Contract.ts

```diff
@@ -81,14 +96,71 @@ import {
   swapFeeGate
 } from "../fees/dexLaunchpadFeeGate";
 import { GalaChainContext } from "../types";
-import { GalaContract } from "./GalaContract";
-import { EVALUATE, Evaluate, GalaTransaction, Submit } from "./GalaTransaction";
+import { BatchWriteLimitExceededError, GalaContract } from "./GalaContract";
+import { getApiMethod } from "./GalaContractApi";
+import { EVALUATE, Evaluate, GalaTransaction, SUBMIT, Submit } from "./GalaTransaction";
 
 export class DexV3Contract extends GalaContract {
   constructor() {
     super("DexV3Contract", version);
   }
 
+  @GalaTransaction({
+    type: SUBMIT,
+    in: BatchDto,
+    out: "object",
+    description: "Submit a batch of transactions",
+    allowedOrgs: [process.env.CURATOR_ORG_MSP ?? "CuratorOrg"],
+    verifySignature: true
+  })
+  public async BatchSubmit(ctx: GalaChainContext, batchDto: BatchDto): Promise<GalaChainResponse<unknown>[]> {
+    // Check if the calling user is authorized to submit batches
+    const batchAuthorizations = await fetchBatchSubmitAuthorizations(ctx);
+    if (!batchAuthorizations.isAuthorized(ctx.callingUser)) {
+      throw new UnauthorizedError(
+        `CallingUser ${ctx.callingUser} is not authorized to submit batches. ` +
+          `Authorized users: ${batchAuthorizations.getAuthorizedAuthorities().join(", ")}`
+      );
+    }
+
+    const responses: GalaChainResponse<unknown>[] = [];
+
+    const softWritesLimit = batchDto.writesLimit ?? BatchDto.WRITES_DEFAULT_LIMIT;
+    const writesLimit = Math.min(softWritesLimit, BatchDto.WRITES_HARD_LIMIT);
+    let writesCount = ctx.stub.getWritesCount();
+
+    for (const [index, op] of batchDto.operations.entries()) {
+      // Use sandboxed context to avoid flushes of writes and deletes, and populate
+      // the stub with current writes and deletes.
+      const sandboxCtx = ctx.createReadOnlyContext(index);
+      sandboxCtx.stub.setWrites(ctx.stub.getWrites());
+      sandboxCtx.stub.setDeletes(ctx.stub.getDeletes());
+
+      // Execute the operation. Collect both successful and failed responses.
+      let response: GalaChainResponse<unknown>;
+      try {
+        if (writesCount >= writesLimit) {
```

Comment:
I usually expect that I'm allowed to hit a "limit", but not exceed it, so consider using strictly greater than.